name-computer/map.jinja

```diff
@@ -23,6 +26,15 @@ RedHat:
       - hostname: {{ name }}
       - apply_hostname: True
       - retain_settings: True
+
+  Update /etc/hosts:
+    host.present:
+      - ip: {{ salt.network.interface(salt.network.default_route('inet')[0]['interface'])[0]['address'] }}
```

Comment:
The salt version pinned in appveyor.yml does not have a `default_route` function... Pretty sure I actually contributed that function, and we're using 2018.3.4 anyway now, so go ahead and update the version pin and see if that clears up the failure:

* https://repo.saltstack.com/windows/Salt-Minion-2018.3.4-Py2-AMD64-Setup.exe

Reply:
Actually I implemented only the [`get_route` function](https://github.com/saltstack/salt/pull/41103)... But, the `default_route` concept can be a bit arbitrary anyway. Maybe replace this whole thing with:

```suggestion
      - ip: {{ salt.network.get_route('192.0.0.8')['source'] }}
```

`192.0.0.8/32` is the [dummy IPv4 address](https://www.iana.org/assignments/iana-ipv4-special-registry/iana-ipv4-special-registry.xhtml), reserved, it ought never be local and never actually be assigned, so the idea is that this checks the route table to see which interface would be used to send to this dummy address and returns the source IP of that interface. This way we have a better idea of what algorithm exactly is being used to determine the default route...

```
> C:\salt\salt-call.bat --local network.get_route 192.0.0.8
local:
    ----------
    destination:
        192.0.0.8
    gateway:
        192.168.84.1
    interface:
        Wi-Fi
    source:
        192.168.84.31
```

Reply:
Totally not following

Reply:
Well, you can rework the approach to conditionally execute this function based on os-type, or you can use a function that already has broader OS support (and which does not require hardcoding the 0 indices).

Reply:
Probably need to do the former. The formula (notionally) needs to be portable beyond a "single-NIC host" context. While, even on a multi-interface system, no one should have a static-route that includes 192.0.0.0/N, grabbing the zero-index route is about as close as Linux gets for letting you reliably determine the default interface.

Reply:
/shrug I don't think there is a default interface either, and can't even say for sure that salt is going to return the list in the same order every time, so the 0 index isn't necessarily stable anyway. We already use the `get_route` approach for the name-computer pillar, though yeah that is at least user-chosen. I suppose could expose an `ip` pillar option, which we could set the same way...

Reply:
Fuckit - going to KISS it, I guess:
1) The proposed method is shorter (doesn't blow out margins)
2) People likely to use this on multi-interface systems is, I'd wager, incredibly-small
3) The number of people likely to use this with a multi-interface system _and_ that are violating networking-standards by setting a static-route (and on a _secondary_ interface) that would encompass that reserved IP-range is presumably vanishingly-small